packages/web3-eth/test/integration/rpc.test.ts

```diff
@@ -0,0 +1,402 @@
+/*
+This file is part of web3.js.
+
+web3.js is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+web3.js is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
+*/
+import WebSocketProvider from 'web3-providers-ws';
+import { FMT_BYTES, FMT_NUMBER } from 'web3-common';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { Contract, decodeEventABI } from 'web3-eth-contract';
+import { hexToNumber } from 'web3-utils';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { AbiEventFragment } from 'web3-eth-abi';
+import { ReceiptInfo, Web3Eth } from '../../src';
+
+import {
+	getSystemTestAccounts,
+	getSystemTestBackend,
+	getSystemTestProvider,
+	itIf,
+} from '../fixtures/system_test_utils';
+import { basicContractAbi, basicContractByteCode } from '../shared_fixtures/sources/Basic';
+import { toAllVariants } from '../shared_fixtures/utils';
+import { sendFewTxes } from './helper';
+
+const mapFormatToType: { [key: string]: string } = {
+	[FMT_NUMBER.NUMBER]: 'number',
+	[FMT_NUMBER.HEX]: 'string',
+	[FMT_NUMBER.STR]: 'string',
+	[FMT_NUMBER.BIGINT]: 'bigint',
+};
+const eventAbi: AbiEventFragment = basicContractAbi.find((e: any) => {
+	return e.name === 'StringEvent' && (e as AbiEventFragment).type === 'event';
+})! as AbiEventFragment;
+describe('rpc', () => {
+	let web3Eth: Web3Eth;
+	let accounts: string[] = [];
+	let clientUrl: string;
+
+	let contract: Contract<typeof basicContractAbi>;
+	let deployOptions: Record<string, unknown>;
+	let sendOptions: Record<string, unknown>;
+
+	beforeAll(async () => {
+		clientUrl = getSystemTestProvider();
+		accounts = await getSystemTestAccounts();
+		web3Eth = new Web3Eth(clientUrl);
+
+		contract = new Contract(basicContractAbi, undefined, {
+			provider: clientUrl,
+		});
+
+		deployOptions = {
+			data: basicContractByteCode,
+			arguments: [10, 'string init value'],
+		};
+
+		sendOptions = { from: accounts[0], gas: '1000000' };
+
+		contract = await contract.deploy(deployOptions).send(sendOptions);
+	});
+	afterAll(() => {
+		if (clientUrl.startsWith('ws')) {
+			(web3Eth.provider as WebSocketProvider).disconnect();
+		}
+	});
+
+	describe('methods', () => {
```

Comment:
As annoying as this change is going to be, to keep consistency, we should split each method into it's own file like [rpc_method_wrappers](https://github.com/ChainSafe/web3.js/tree/4.x/packages/web3-eth/test/unit/rpc_method_wrappers) and [rpc_methods](https://github.com/ChainSafe/web3.js/tree/wyatt/4.x/4993-web3-eth-test-refactors/packages/web3-eth/test/unit/rpc_methods). I'm not so sure this is the best approach, but it was agreed upon, what `1.x` does, and it makes it easier to debug specific failing tests instead of have to run the entire suite